### ISSUE TITLE
Manually restore from project.json file?

### ISSUE THREAD

**Issue by victorhooi** [mpeterson2/save-session#71]
I recently did an Atom upgrade to to 0.181.0.

After a restart, my session did not appear to restore.

I went into my `~/.atom/package/save-session/projects` directory, and I was able to find the `projects.json` file for my last session. Looking through this, I can see the text from several of my unsaved buffers.

I've tried restarting Atom a few times, but it doesn't seem to restore this session.

I have made a backup of this `projects.json` file.

My next question is, is there some way I can force save-session to restore the session from this file? Or even just a way to export all of the open buffers in `projects.json` into files?

**Comment by mpeterson2** [comment]
There's not currently an easy way to do it. It is all just stored in json, so you could parse through it if you wanted to. It *should* restore it whenever you relaunch through, so that's kinda weird.

Check what your project setting is. It may be empty for some reason, and if it is, just set it to the path to get to your .project file.

**Comment by rekoilgzs** [comment]
I had this same issue and figured out how to fix it. It's not relocating your project path but relocating the addon's Data Save Folder. For some reason with the Atom update, it changed my folder path from:
C:\Users\X\C\Users\X\project.json (or something of the sort) simply to C:\Users\X\undefined\project.json . I simply copied the original file over top of the new file it had created in the "undefined" path and voila, all my files were back.

Hope this helps.

**Comment by victorhooi** [comment]
Hmm, I can confirm that I copied the `project.json` file over to:

```
/Users/victorhooi/.atom/packages/save-session/projects/undefined
```

And it restored it from that file - not sure what's going on here.

Also, kudos to mpeterson2 on this project - but hopefully this feature will make it into Atom core itself soon:

https://github.com/atom/atom/issues/1603

They also mentioned another plugin there, https://github.com/danielmahon/atom-open-last-project, which apparently leverages some in-built functionality of Atom, will be interesting to compare this.

mpeterson2 - have you seen this?

**Comment by mpeterson2** [comment]
@victorhooi No, I haven't seen that plugin yet. I took a short look at it, and I may be missing something, but it doesn't seem to do anything quite yet. Maybe they can find a better way to do this.

On the other hand, I just saw that they may be putting this feature into the Atom core for 1.0 [here](https://github.com/atom/atom/issues/1603#issuecomment-71072551), which would be pretty awesome.
